pkg/writer/sonar.go

```diff
@@ -47,8 +49,13 @@ func (s *Sonar) Write(mets []*dto.MetricFamily) error {
 		}
 
 	}
-
-	return s.client.Flush()
+	err := s.client.Flush()
+	httpError, ok := err.(*tsclient.UnexpectedHTTPStatusError)
+	if !s.firstWriteSent && ok && httpError.StatusCode == 429 {
+		err = nil
+	}
+	s.firstWriteSent = true
```

Comment:
This is racey and should be wrapped in a sync.Once. Rather than using a boolean value (which is not thread safe). You should create a field on Sonar called `onceIgnoreError sync.Once` and you should change this block to be something like:

```
err := s.client.Flush()
s.onceIgnoreError.Do(func() {
    httpError, ok := err.(*tsclient.UnexpectedHTTPStatusError)
    if ok && httpError.StatusCode == 429 {
        err = nil
    }
})
return err